tests/platform_tests/test_cont_warm_reboot.py

```diff
@@ -100,6 +100,7 @@ def reboot_and_check(self):
         # Wait until uptime reaches allowed value
         self.wait_until_uptime()
         # Perform additional post-reboot health-check
+        self.verify_no_coredumps()
```

Comment:
How to handle the case when there are already core files before test running?

Reply:
I have added a check now to test pre-existing coredumps. I considered deleting (clean-up) the cores before starting the test. But this can prove harmful for core analysis and debugging the test which had created these cores. So, the test would just fail if there are preexisting cores.

Reply:
I don't think it's a good idea that fail the test immediately if core files exist, since there may be other program crashed (like python). 
I have two possible ideas to WA
1. Count core files before and after test run;
2. Filter core files according to creating time.
BTW, we should be ware that there are both UPLOADED_xxx.core.gz and core dump files under /var/core at a certain point.

Reply:
I understand your concern. But, should we at all go ahead with this test if a core is found preexisting within the device?
My concern - the test supports installing images at the beginning and between reboots. If a new installation is to be done (current image != new image), the cores will be deleted, and this may cause of loss of information for debug.
Should we make a backup of these core files before starting the test?

Reply:
IMO, the test should run even there have been core files. After all, we are attempting to verify continuous_reboot, and shouldn't be affected by pre-running cases. 
And I don't think it's necessary to backup core files before upgrading for now. Two reasons,  1. the core files are probably uploaded 2. Backup and restore core files means a lot of code and logic, which is not worth doing.

Reply:
I agree that there is no harm with proceeding with the test if the core files are uploaded.
I was only thinking about the cores which are not uploaded, and get deleted due to this test installing a new image. Losing cores may lead to lost RCA for the test which originally caused these cores. You are right, this test by itself isn't affected by pre-running test cases/cores, and can go ahead without an issue.

Reply:
When I start this test locally, I just manually clean the `/var/core/` if there are any core files before the test. So local version of the testcase isn't a problem anyway.
The problem is when this test runs on nightly, along with other tests. But, there we do not install a new image on the DUT. So, the cores will not be deleted anyway.
I will let the test continue to run even if core files are present, and filter the new cores (if present) to decide to fail the test.

Reply:
Thanks @bingwang-ms for the suggestions. I have made the new changes now.